### DISCUSSION TITLE
Problem with .net references after fresh install?

### DISCUSSION THREAD

**Discussion by ariscol** [dataplat/dbatools#8771] [Q&A] [answered]
So I just did a fresh install of:

- Windows 11
- SQL Server 2022 (Developer Edition)
- Visual Studio 2022 (Professional Edition)
- Powershell Core (x64) - 7.3.2
- Chocolatey

After that, I tried to install dbatools (choco install dbatools), which ran without issue, but when I tried to run a Restore-DbaDatabase command, I got the following error:

WARNING: [12:02:00][Restore-DbaDatabase] Failure | Could not load type 'Microsoft.SqlServer.Server.SqlContext' from assembly 'System.Data, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.

Seems like some wires got crossed somewhere, a mixup between the .net core System.Data and the old (full) .net framework, perhaps? Does anyone have any idea where I might've gone wrong, or what I could try to further diagnose the issue?

I have tried uninstalling with chocolatey, and reinstalling again with Install-Module (from PSGallery). I've tried completely uninstalling PS7 and reinstalling. I'm not sure what else to try. I've verified that the System.Data assembly that is loaded is coming from C:\Program Files\PowerShell\7\System.Data.dll, so maybe that is wrong?

**Comment by wsmelton** [comment]
Our module requires .NET 4.6 to be installed at this time. Windows 11 by default won't have that version installed.

**Reply by ariscol** [reply]
I ticked the boxes for .Net Framework 4.6.2 SDK & targeting pack with Visual Studio. That wouldn't cut it?

**Reply by ariscol** [reply]
hmm, actually the full .net 4 framework installer informed that it is already installed. (as are 2.1, 3.1, 5, 6, 7)

**Reply by wsmelton** [reply]
The SDK is only for compiling applications. It won't install the runtime on the device that I recall. You would need to verify what runtime versions are available to PowerShell.

If you ran the command provided in the issue template and it only returned 4.8 then that is the only runtime installed.

**Comment by wsmelton** [comment, marked as answer]
Ah forgot about the issue is with SQL Server 2022 installation. 

Ref #8673

**Reply by ariscol** [reply]
Dangit, I'm sorry I missed that. I guess I was just searching through discussions and not issues.

Reading that issue and a couple of your recent discussions, it looks like the release of 2.0 is imminent and clearly resolves this issue. I eagerly await :)
